trie/branches.py

```diff
@@ -0,0 +1,167 @@
+
+
+def _if_branch_exist(db, node_hash, key_prefix):
+    # Empty trie
+    if node_hash == BLANK_HASH:
+        return False
+    nodetype, left_child, right_child = parse_node(db[node_hash])
+    if nodetype == LEAF_TYPE:
+        if key_prefix:
+            return False
+        return True
+    elif nodetype == KV_TYPE:
+        if not key_prefix:
+            return True
+        if len(key_prefix) < len(left_child):
+            if key_prefix == left_child[:len(key_prefix)]:
+                return True
+            return False
+        else:
+            if key_prefix[:len(left_child)] == left_child:
+                return _if_branch_exist(db, right_child, key_prefix[len(left_child):])
+            return False
+    elif nodetype == BRANCH_TYPE:
+        if not key_prefix:
+            return True
+        if key_prefix[:1] == BYTE_0:
+            return _if_branch_exist(db, left_child, key_prefix[1:])
+        else:
+            return _if_branch_exist(db, right_child, key_prefix[1:])
+
+
+def get_branch(db, root_hash, key):
+    """
+    Get a long-format Merkle branch
+    """
+    validate_is_bytes(key)
+
+    return _get_branch(db, root_hash, encode_to_bin(key))
+
+
+def _get_branch(db, node_hash, keypath):
+    if node_hash == BLANK_HASH:
+        return []
+    node = db[node_hash]
+    nodetype, left_child, right_child = parse_node(node)
+    if nodetype == LEAF_TYPE:
+        if not keypath:
+            return [node]
+        else:
+            raise InvalidKeyError("Key too long")
+    elif nodetype == KV_TYPE:
+        if not keypath:
+            raise InvalidKeyError("Key too short")
+        if keypath[:len(left_child)] == left_child:
+            return [node] + _get_branch(
+                db,
+                right_child,
+                keypath[len(left_child):]
+            )
+        else:
+            raise InvalidKeyError("Key does not exist")
+    elif nodetype == BRANCH_TYPE:
+        if not keypath:
+            raise InvalidKeyError("Key too short")
+        if keypath[:1] == BYTE_0:
+            return [node] + _get_branch(db, left_child, keypath[1:])
+        else:
+            return [node] + _get_branch(db, right_child, keypath[1:])
+
+
+def verify_branch(branch, root_hash, key, value):
+    validate_is_bytes(key)
+
+    db = {keccak(node): node for node in branch}
+    assert BinaryTrie(db=db, root_hash=root_hash).get(key) == value
+    return True
+
+
+def get_trie_nodes(db, node_hash):
+    """
+    Get full trie of a given root node
+    """
+    if node_hash in db:
+        node = db[node_hash]
+    else:
+        return []
+    nodetype, left_child, right_child = parse_node(node)
+    if nodetype == KV_TYPE:
+        return [node] + get_trie_nodes(db, right_child)
+    elif nodetype == BRANCH_TYPE:
+        return [node] + get_trie_nodes(db, left_child) + get_trie_nodes(db, right_child)
+    elif nodetype == LEAF_TYPE:
+        return [node]
```

Comment:
Can you add a final `else` with a `raise Exception("Invariant: unreachable code path")` just for good measure.